mgmt_asset/demo/demo.xml

```diff
@@ -1,7 +1,14 @@
 <odoo>
 
-    <record id="demo_asset0" model="mgmt.asset">
-        <field name="name">Asset 0</field>
+    <record id="demo_inventory_category_servers" model="maintenance.equipment.category">
+        <field name="name">Servers</field>
     </record>
 
+    <!-- <record id="demo_inventory_server" model="maintenance.equipment"> -->
+    <!--     <field name="name">Odoo-Server</field> -->
+    <!--     <field name="category_id" ref="demo_inventory_category_servers"/> -->
+    <!--     <field name="equipment_assign_to">employee</field> -->
```

Comment:
This should be given as a default.

Reply:
@janikvonrotz sorry, but I don't quite understand how to make this work. The `equipment_assign_to` is a selection field: 
![image](https://github.com/user-attachments/assets/89bd5b32-b391-4ee4-8eb0-d2d8d0d0918d)
I provide a value "employee" to this field, but it is invalid. I can create same record manually in the UI and it does not require assigning to a specific user. It is absolutely not clear for me why this demo data would not work and what needs to be changed. I understand that extending `maintenance.equipment` is not an option either. Please give some advice on how to resolve this as I am already spending way too much time on this one

Reply:
@laszlokr Did you check the code of maintenance? The field has a default value: https://github.com/odoo/odoo/blob/65d4ff2ed145db978843d7f2f04263956c4c0d96/addons/hr_maintenance/models/equipment.py#L13
Therefore I expect that you don't have to define this field.

What is the actual issue with the demo data? Are you sure you did not get side tracked on this issue?

Reply:
@janikvonrotz I haven't checked that it has a default value, but it does not change much. Here is the error when the field is not defined:
```
psycopg2.errors.NotNullViolation: null value in column "equipment_assign_to" violates not-null constraint
DETAIL:  Failing row contains (12, 1, null, null, null, null, null, null, 6, null, null, 1, 1, null, null, null, null, 2024-09-18, null, null, null, {"en_US": "Odoo-Server"}, null, t, 2024-09-18 10:01:30.55465, 2024-09-18 10:01:30.55465, null, null, null, null).

    raise ParseError('while parsing %s:%s, somewhere inside\n%s' % (
odoo.tools.convert.ParseError: while parsing /home/laszlo/work/mintsystem/Odoo-Build/addons/management-system/mgmt_asset/demo/demo.xml:7, somewhere inside
<record id="demo_inventory_server" model="maintenance.equipment">
        <field name="name">Odoo-Server</field>
        <field name="category_id" ref="demo_inventory_category_servers"/>
    </record>
```

When the field is defined but selection not specified:
```
    raise ValueError("Invalid field %r on model %r" % (key, self._name))
ValueError: Invalid field 'equipment_assign_to' on model 'maintenance.equipment'

odoo.tools.convert.ParseError: while parsing /home/laszlo/work/mintsystem/Odoo-Build/addons/management-system/mgmt_asset/demo/demo.xml:7, somewhere inside
<record id="demo_inventory_server" model="maintenance.equipment">
        <field name="name">Odoo-Server</field>
        <field name="category_id" ref="demo_inventory_category_servers"/>
        <field name="equipment_assign_to"/>
    </record>
```

When the field is defined and selection is specified, same goes for any other non default selection:
```
    raise ValueError("Invalid field %r on model %r" % (key, self._name))
ValueError: Invalid field 'equipment_assign_to' on model 'maintenance.equipment'

odoo.tools.convert.ParseError: while parsing /home/laszlo/work/mintsystem/Odoo-Build/addons/management-system/mgmt_asset/demo/demo.xml:7, somewhere inside
<record id="demo_inventory_server" model="maintenance.equipment">
        <field name="name">Odoo-Server</field>
        <field name="category_id" ref="demo_inventory_category_servers"/>
        <field name="equipment_assign_to">employee</field>
    </record>
```

Not sure where I could get side tracked, but of course that's also possible. I just tried all options there are and still cannot make this piece of demo data load